ale_linters/proto/protoc-gen-lint.vim

```diff
@@ -0,0 +1,11 @@
+" Author: Jeff Willette <jrwillette88@gmail.com>
+" Description: run the protoc-gen-lint plugin for the protoc binary
+
+call ale#linter#Define('proto', {
+\   'name': 'protoc-gen-lint',
+\   'output_stream': 'stderr',
+\   'lint_file': 1,
+\   'executable': 'protoc',
+\   'command': 'protoc -I $(dirname %s) --lint_out=. %s',
```

Comment:
You've got the right idea about using `'command_callback' here. What you should do is write a callback with `'command_callback'` and add `ale#Escape(expand('#' . a:buffer . ':p:h'))` into the command to add the escaped directory name. That will work everywhere. Then add a test for the command callback in the `test/command_callback` directory. There are a few similar tests.